### ISSUE TITLE
Warnings from the firebase_analytics plugin on Android: FirebaseAnalyticsPlugin.java uses unchecked or unsafe operations

### ISSUE THREAD

**Issue by mulderpf** [flutter/flutter#25402]
## Steps to Reproduce

1. Create a new project
2. Add a dependency for `firebase_analytics: ^1.0.6`
3. `flutter run`

## Logs

```
[+1999 ms] > Task :firebase_analytics:compileDebugJavaWithJavac
[   +4 ms] Note:
C:\code\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_analytics-1.0.6\android\src\main\java\io\flutter\plugins\fir
ebaseanalytics\FirebaseAnalyticsPlugin.java uses unchecked or unsafe operations.
[   +3 ms] Note: Recompile with -Xlint:unchecked for details.
```

```
Analyzing analyticsissue...
No issues found! (ran in 13.7s)
```

```
[√] Flutter (Channel master, v1.1.2-pre.36, on Microsoft Windows [Version 10.0.17134.471], locale en-GB)
    • Flutter version 1.1.2-pre.36 at C:\code\flutter
    • Framework revision c5251cdc02 (8 hours ago), 2018-12-14 16:09:17 -0800
    • Engine revision 101b27da7a
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at C:\Users\mulde\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[√] Android Studio (version 3.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 30.0.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[√] VS Code (version 1.29.1)
    • VS Code at C:\Users\mulde\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 2.21.1

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!
```

**Comment by danwulff** [comment]
@zoechi @mulderpf 
### I'm also seeing this with v2.0.3. See below for more info and a reproduction:

After adding the firebase_analytics plugin to a new flutter application (Stable Channel, migrated to androidX) a "uses unchecked or unsafe operations" warning is present when running a build (Android only).

The warning presented:

> Note: /Users/dwulff/.pub-cache/hosted/pub.dartlang.org/firebase_analytics-2.0.3/android/src/main/java/io/flutter/plugins/firebaseanalytics/FirebaseAnalyticsPlugin.java uses unchecked or unsafe operations.
> Note: Recompile with -Xlint:unchecked for details. 

## Steps to Reproduce

### option 1)
- clone https://github.com/danwulff/flutter_firebase_analytics_unsafe_warning.git
- run `flutter run`

### option 2)
(using stable channel, v1.2.1)
- run `flutter create ___`
- migrate to AndroidX using Android Studio
- add `firebase_analytics: 2.0.3` to pubspec.yaml
- run `flutter run` in a terminal

## Logs

flutter run
```
$ flutter run
Using hardware rendering with device Android SDK built for x86. If you get graphics artifacts, consider enabling software rendering with
"--enable-software-rendering".
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...                                              0.7s
Resolving dependencies...                                           1.3s
Note: /Users/dwulff/.pub-cache/hosted/pub.dartlang.org/firebase_analytics-2.0.3/android/src/main/java/io/flutter/plugins/firebaseanalytics/FirebaseAnalyticsPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.                      
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                        15.6s
Built build/app/outputs/apk/debug/app-debug.apk.
Installing build/app/outputs/apk/app.apk...                         1.4s
D/        (13864): HostConnection::get() New Host Connection established 0xebc66640, tid 13884
D/EGL_emulation(13864): eglMakeCurrent: 0xec0f6040: ver 3 0 (tinfo 0xebc3f540)
Syncing files to device Android SDK built for x86...             1,612ms

🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
An Observatory debugger and profiler on Android SDK built for x86 is available at: http://127.0.0.1:57725/
For a more detailed help message, press "h". To detach, press "d"; to quit, press "q".
```

flutter doctor -v
```
[✓] Flutter (Channel stable, v1.2.1, on Mac OS X 10.14.3 18D109, locale en-US)
    • Flutter version 1.2.1 at /Users/dwulff/development/flutter
    • Framework revision 8661d8aecd (5 weeks ago), 2019-02-14 19:19:53 -0800
    • Engine revision 3757390fa4
    • Dart version 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/dwulff/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/dwulff/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 33.3.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] VS Code (version 1.32.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.24.0

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!

```

**Comment by kroikie** [comment]
@mulderpf

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/544. Any further collaboration will be done there.
